cmd/broadcast/utils.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"bufio"
+	"encoding/hex"
+	ida "github.com/harmony-one/libunison/internal/ida/coopcast"
+	"io"
+	"log"
+	"math/rand"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const pubKeySize int = 20
+
+// Entry is a single config of a node.
+type Entry struct {
+	Sid     string
+	IP      string
+	TCPPort string
+	UDPPort string
+	PubKey  string
+	Role    string
+}
+
+// Config is a struct containing multiple Entry of all nodes.
+type Config struct {
+	config []Entry
+}
+
+// NewConfig returns a pointer to a Config.
+func NewConfig() *Config {
+	config := Config{}
+	return &config
+}
+
+// GetPeerInfo returns the selfPeer, peerList, allPeers from config instance, which used to create node instance
+func (config *Config) GetPeerInfo() (ida.Peer, []ida.Peer, []ida.Peer) {
+	var allPeers []ida.Peer
+	var peerList []ida.Peer
+	var selfPeer ida.Peer
+	for _, entry := range config.config {
+		sid, err := strconv.Atoi(entry.Sid)
+		if err != nil {
+			log.Printf("cannot convert sid")
+		}
+		peer := ida.Peer{IP: entry.IP, TCPPort: entry.TCPPort, UDPPort: entry.UDPPort, PubKey: entry.PubKey, Sid: sid}
+		if entry.Role == "0" {
+			selfPeer = peer
+		} else if entry.Role == "1" {
+			peerList = append(peerList, peer)
+			allPeers = append(allPeers, peer)
+		} else {
+			allPeers = append(allPeers, peer)
+		}
+	}
+	return selfPeer, peerList, allPeers
+}
+
+// ReadConfigFile parses the config file and return a 2d array containing the file data
+func (config *Config) ReadConfigFile(filename string) error {
+	file, err := os.Open(filename)
+	if err != nil {
+		log.Fatal("Failed to read config file ", filename)
+		return err
+	}
+	defer file.Close()
+	fscanner := bufio.NewScanner(file)
+
+	result := []Entry{}
+	for fscanner.Scan() {
+		p := strings.Split(fscanner.Text(), " ")
+		entry := Entry{p[0], p[1], p[2], p[3], p[4], p[5]}
+		result = append(result, entry)
+	}
+	config.config = result
+	return nil
+}
+
+// GenerateConfigFromGraph generate config files from graph config file using adjacent map definition of a graph
+func GenerateConfigFromGraph(graphfile string) {
+	file, err := os.Open(graphfile)
+	if err != nil {
+		log.Fatal("Failed to read config file ", graphfile)
+		return
+	}
+
+	defer file.Close()
+	fscanner := bufio.NewScanner(file)
+	var n int
+	fscanner.Scan()
+	n, err = strconv.Atoi(fscanner.Text())
+	if err != nil {
+		log.Printf("not able to convert to number of nodes")
+	}
+	pubkeys, tcps, udps := initConfig(n)
+
+	for fscanner.Scan() {
+		p := strings.Split(fscanner.Text(), " ")
+		writeGraphRelationToConfig(p, n, pubkeys, tcps, udps)
+	}
+}
+
+func initConfig(n int) (map[int][]byte, []int, []int) {
+	filename := "configs/config_allpeers.txt"
+	f, err := os.Create(filename)
+	if err != nil {
+		log.Printf("cannot create file: %v", filename)
+	}
+	defer f.Close()
+
+	rand.Seed(time.Now().UnixNano())
```

Comment:
RNGs should be seeded once at the beginning of the program.  Repeated seeding, especially with predictable values such as timestamp, may degrade and/or bias randomness.